### PR TITLE
feat: search products

### LINKED ISSUE #14: Search Products | Searchbar
Search products in searchbar and it will filter the products based on the search input.

### PR DESCRIPTION
**Pull request by tushartiwari7**
Added Product Search Functionality.
Now user can search products by the title and it will filter the products based on entered value in search bar.

search

### CODE REVIEW

**Review by harshalini — APPROVED** [review]
Overall code is good, merge it

**Review by dikshadhote — APPROVED** [review]
Overall the code is well structured.
Few points that I observed in preview:

- Footer is missing
- The text in card is going above images.so, use proper padding margin.
- The categories text just seems a text on page rather than something clickable when user see's it for first time. give some box-shadow to make it look as CTA

image
- Please give border radius to login button

**Review by neetukumari4858 — APPROVED** [review]
code is well structured. it is good to merge

File: src/components/Header/Header.jsx [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
-import { useUser } from "../../context";
+import { useUser, useProducts } from "../../context";
 
 export const Header = () => {
   const { user } = useUser();
+  const { dispatch } = useProducts();
+  const navigator = useNavigate();
+
+  const onSearch = (e) => {
+   if(window.location.pathname !== '/products')
```
Comment (JantuDeb) [inline src/components/Header/Header.jsx]: You could have used `useLocation()` hooks from react router.

File: src/components/Header/Header.jsx [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+      navigator("/products");
+    return dispatch({ type: "PRODUCTS_SEARCH", payload: e.target.value });
```
Comment (JantuDeb) [inline src/components/Header/Header.jsx]: You don't need to return `dispatch()` function

**Review by JantuDeb — APPROVED** [review]

File: src/components/Header/Header.jsx (hunk 1 above)
Comment (tushartiwari7) [reply src/components/Header/Header.jsx]: Thanks, didn't knew that exists.

Comment (tushartiwari7) [reply src/components/Header/Header.jsx]: done

File: src/components/Header/Header.jsx (hunk 2 above)
Comment (tushartiwari7) [reply src/components/Header/Header.jsx]: done

**Pull request merged**